internal/server/server.go

```diff
@@ -338,13 +338,24 @@ func (s *Server) handleLogin() http.HandlerFunc {
 			return
 		}
 		session.Values["nonce"] = nonce
+		// generate a PKCE code verifier
+		codeVerifier := oauth2.GenerateVerifier()
+		codeChallenge := oauth2.S256ChallengeFromVerifier(codeVerifier)
+
+		// store code_verifier for token exchange
+		session.Values["code_verifier"] = codeVerifier
 
 		if err := session.Save(r, w); err != nil {
 			logAndError(w, http.StatusInternalServerError, err, "error saving session")
 			return
 		}
 
-		http.Redirect(w, r, s.oauth2Config.AuthCodeURL(randomStr, oidc.Nonce(nonce)), http.StatusSeeOther)
+		redirectURL := s.oauth2Config.AuthCodeURL(state,
+			oidc.Nonce(nonce),
+			oauth2.SetAuthURLParam("code_challenge", codeChallenge),
+			oauth2.SetAuthURLParam("code_challenge_method", "S256"))
```

Comment:
Use https://pkg.go.dev/golang.org/x/oauth2#S256ChallengeOption.

If this is live in the test cluster I don't see the code_challenge_method query parameter when submitting the request to dex. Since dex doesn't allow us to disable the plaintext challenge method, we have an explicit check for the code_challenge_method query parameter value S256 so it MUST be included as a query parameter and not as part of the POST body.

Reply:
Reverted to challengeOption